AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift

```diff
@@ -618,29 +620,310 @@ class ItemListProcessorTests: AuthenticatorTestCase { // swiftlint:disable:this
         subject.didCompleteAutomaticCapture(captureCoordinator.asAnyCoordinator(), key: key)
+        waitFor(!coordinator.alertShown.isEmpty)
+
+        let alert = try XCTUnwrap(coordinator.alertShown.first)
+        XCTAssertEqual(alert.alertActions.count, 2)
+        let saveLocallyOption = try XCTUnwrap(alert.alertActions.first)
+        XCTAssertEqual(saveLocallyOption.title, Localizations.saveHere)
+        await saveLocallyOption.handler?(saveLocallyOption, [])
+
         var dismissAction: DismissAction?
         if case let .dismiss(onDismiss) = captureCoordinator.routes.last {
             dismissAction = onDismiss
         }
         XCTAssertNotNil(dismissAction)
         dismissAction?.action()
+
+        try await Task.sleep(nanoseconds: 500_000)
         waitFor(!authItemRepository.addAuthItemAuthItems.isEmpty)
+        try await Task.sleep(nanoseconds: 500_000)
         waitFor(subject.state.loadingState != .loading(nil))
-        guard let item = authItemRepository.addAuthItemAuthItems.first
-        else {
-            XCTFail("Unable to get authenticator item")
-            return
+        let item = try XCTUnwrap(authItemRepository.addAuthItemAuthItems.first)
+        XCTAssertEqual(item.name, "")
+        XCTAssertEqual(item.totpKey, String.base32Key)
+    }
+
+    /// `didCompleteAutomaticCapture` success when the user has opted to be asked by default and
+    /// chooses the save locally option.
+    func test_didCompleteAutomaticCapture_hasSeenPrompt_noneSaveToBitwardenChosen() async throws {
+        configService.featureFlagsBool[.enablePasswordManagerSync] = true
+        application.canOpenUrlResponse = true
+        appSettingsStore.hasSeenDefaultSaveOptionPrompt = true
+        appSettingsStore.defaultSaveOption = .none
+        let key = String.base32Key
+        let keyConfig = try XCTUnwrap(TOTPKeyModel(authenticatorKey: key))
+        totpService.getTOTPConfigResult = .success(keyConfig)
+        authItemRepository.itemListSubject.value = [ItemListSection.fixture()]
+        let captureCoordinator = MockCoordinator<AuthenticatorKeyCaptureRoute, AuthenticatorKeyCaptureEvent>()
+        subject.didCompleteAutomaticCapture(captureCoordinator.asAnyCoordinator(), key: key)
+        waitFor(!coordinator.alertShown.isEmpty)
+
+        let alert = try XCTUnwrap(coordinator.alertShown.first)
+        XCTAssertEqual(alert.alertActions.count, 2)
+        let saveLocallyOption = try XCTUnwrap(alert.alertActions[1])
+        XCTAssertEqual(saveLocallyOption.title, Localizations.takeMeToBitwarden)
+        await saveLocallyOption.handler?(saveLocallyOption, [])
+
+        var dismissAction: DismissAction?
+        if case let .dismiss(onDismiss) = captureCoordinator.routes.last {
+            dismissAction = onDismiss
+        }
+        XCTAssertNotNil(dismissAction)
+        dismissAction?.action()
+
+        try await Task.sleep(nanoseconds: 500_000)
```

Comment:
Bitwarden PM app has a waitForAsync that maybe you can use instead of the sleeps

Reply:
Good call. 👍 I've brought it over to Authenticator in the latest and replaced all the `sleep`s.